implementations/rust/ockam/ockam_api/src/kafka/portal_worker.rs

```diff
@@ -239,17 +258,19 @@ impl KafkaPortalWorker {
     /// used for requests.
     pub(crate) async fn start(
         context: &mut Context,
+        secure_channel_controller: Arc<dyn KafkaSecureChannelController>,
+        uuid_to_name: TopicUuidMap,
         inlet_map: KafkaInletMap,
     ) -> ockam_core::Result<Address> {
-        let shared_protocol_state = ProtocolState::new();
+        let shared_protocol_state = ProtocolState::new(secure_channel_controller, uuid_to_name);
 
         let inlet_address = Address::random_tagged("KafkaPortalWorker.inlet");
         let outlet_address = Address::random_tagged("KafkaPortalWorker.outlet");
         let disconnect_received = Arc::new(AtomicBool::new(false));
 
         let inlet_worker = Self {
             inlet_map: inlet_map.clone(),
-            shared_protocol_state: shared_protocol_state.clone(),
+            shared_protocol_state: shared_protocol_state.async_try_clone().await?,
```

Comment:
Would it be possible to clone the `inlet_worker` and just set the `receiving` field to `Receiving`? to limit the duplication?

Reply:
They have different parameters, mainly `receiving: Receiving::Requests,` and `receiving: Receiving::Responses` so no